### PR TITLE
Support for python 3.12, pyproject.toml, minor naming cleanups

### PR DESCRIPTION
**Pull request by smjoshiatglobus**
This pull request brings in these changes:
- Replaced `setup.cfg` and `setup.py` with `pyproject.toml`. See issue #54 .
- `README.md` changes
  - Fixed broken example code for `anonymize`. The API requires `delete_private_tags` argument,
  - Fixed broken example code for `anonymize_dataset`. The previous one was throwing exception from pydicom
  - Copied the examples to a new `examples` directory to ensure they work
  - Naming cleanup per PEP 8 style guide
- `anonymizer.py` Changes
  - Replaced deprecated `pkg_resources` with `importlib`
  - Renamed argument to `delete_private_tags` to sync up with the same argument for `anonymize_dataset` and PEP 8 style guide

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,84 +156,58 @@ def main():
...
-  ]
-
-  # Create a readable dataset for pydicom
-  data = pydicom.Dataset()
-
-  # Add each field into the dataset
-  for field in fields:
-    data.add_new(field['id'], field['type'], field['value'])
-
-  anonymize_dataset(data)
```
Comment (pchoisel) [inline README.md:219]: Why did you delete this example ? This can be useful.  
Why not moving it in the examples folder you created ?

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+[project]
+name = "dicom_anonymizer"
+version = "1.0.11"
+authors = [
+  { name="Laurenn Lam", email="laurenn.lam@kitware.com" },
+]
+description = "Program to anonymize dicom files with default and custom rules"
+readme = "README.md"
+requires-python = ">=3.10"
```
Comment (pchoisel) [inline pyproject.toml:9]: Why restrict to 3.10 and higher ? I tested with Python 3.9 and 3.8 and it seems to work

**Comment by pchoisel** [comment]
Thank you very much for your work !

File: README.md (hunk 1 above)
Comment (smjoshiatglobus) [reply README.md:219]: I tried this example and could not get it to work. I can fix the first warning about 'bar' not being valid for `TM`, but have no idea how to fix the error of missing `file_meta` for the dataset.

Here is the output:
```
C:\Users\smjoshi\.virtualenvs\examples-gxyY8XxM\Lib\site-packages\pydicom\valuerep.py:443: UserWarning: Invalid value for VR TM: 'bar'.
  warnings.warn(msg)
Traceback (most recent call last):
  File "C:\development\dicom-anonymizer\examples\anonymize_fields.py", line 33, in <module>
    main()
  File "C:\development\dicom-anonymizer\examples\anonymize_fields.py", line 30, in main
    anonymize_dataset(data)
  File "C:\development\dicom-anonymizer\dicomanonymizer\simpledicomanonymizer.py", line 436, in anonymize_dataset
    action(dataset.file_meta, tag)
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\smjoshi\.virtualenvs\examples-gxyY8XxM\Lib\site-packages\pydicom\dataset.py", line 908, in __getattr__
    return object.__getattribute__(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Dataset' object has no attribute 'file_meta'
```

File: pyproject.toml (hunk 2 above)
Comment (smjoshiatglobus) [reply pyproject.toml:9]: That was my confusion. Per [importlib.metadata](https://docs.python.org/3/library/importlib.metadata.html#module-importlib.metadata) documentation, it was "provisional" until 3.10, so I thought it was not available for versions 3.8 and 3.9. Since these worked, I will update this minimum requirement to 3.8.

Thank you for testing!

Comment (smjoshiatglobus) [reply pyproject.toml]: Fixed in commit b3dc158

File: README.md (hunk 1 above)
Comment (pchoisel) [reply README.md:219]: That should be fixed with https://github.com/KitwareMedical/dicom-anonymizer/pull/61

File: pyproject.toml (hunk 2 above)
Comment (smjoshiatglobus) [reply pyproject.toml]: Thank you! I will merge the main branch into my branch and then clean up my PR.

File: README.md (hunk 1 above)
Comment (smjoshiatglobus) [reply README.md:219]: Thank you! I will pull in the changes into my branch and that should clean up this PR.
